server/src/main/java/com/linecorp/centraldogma/server/internal/api/DtoConverter.java

```diff
@@ -58,29 +60,38 @@ public static RepositoryDto convert(Repository repository) {
                                  repository.creationTimeMillis());
     }
 
-    public static <T> EntryDto<T> convert(Repository repository, Revision revision,
-                                          Entry<T> entry, boolean withContent) {
+    public static <T> EntryDto convert(Repository repository, Revision revision,
+                                       Entry<T> entry, boolean withContent) {
+        return convert(repository, revision, entry, withContent, null);
+    }
+
+    public static <T> EntryDto convert(Repository repository, Revision revision,
+                                       Entry<T> entry, boolean withContent, @Nullable QueryType queryType) {
         requireNonNull(entry, "entry");
         if (withContent && entry.hasContent()) {
-            return convert(repository, revision, entry.path(), entry.type(), entry.content());
+            // TODO(ks-yim): Not sure DtoConverter having business logic and it looks hacky to receive
+            //   queryType to handle JSON_PATH query for JSON5 files.
```

Comment:
I don't have a better solution. 😅 
Let's leave it as a known issue. 👍

Reply:
> Not sure of 'DtoConverter' having business logic and it looks hacky to receive queryType to handle JSON_PATH query for JSON5 files.

Agreed. `QueryType` and `isJson5()` don't match with DTO. How about taking a `Function` rather than adding a specific logic here?
``` java
public static EntryDto convert(Repository repository, Revision revision,
                               Entry<?> entry, boolean withContent, Function<Entry, ?> contentConverter) {
    return convert(repository, revision, entry.path(), entry.type(), contentCoverter.apply(entry));
}
```